types/jsforce/record-result.d.ts

```diff
@@ -1,7 +1,13 @@
 import { SalesforceId } from './salesforce-id';
 
-export interface RecordResult {
+interface ErrorResult {
+    errors: string[];
+    success: false;
+}
+
+interface SuccessResult {
     id: SalesforceId;
-    success: boolean;
-    anys: Object[];
+    success: true;
 }
+
+export type RecordResult = SuccessResult | ErrorResult;
```

Comment:
this definition allows for properties that are irrelevant to the particular case to not even appear in the type once narrowed.